### PR TITLE
Add fuzzing harness to swift-nio-http2.

### PR DESCRIPTION
**Pull request by Lukasa**
Motivation:

Fuzzing is great, we should have a good sensible place to centralise a
harness.

Modifications:

- Add a fuzzing harness.

Result:

There's a common way to do fuzzing.

### CODE REVIEW

File: FuzzTesting/Sources/FuzzHTTP2/main.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+// Sources/protoc-gen-swift/main.swift - Protoc plugin main
+//
+// Copyright (c) 2020 - 2021 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
```
Comment (glbrntt) [inline FuzzTesting/Sources/FuzzHTTP2/main.swift:20]: Intentional? / Should we add an entry to NOTICES.txt?

File: scripts/soundness.sh [hunk 2]
```diff
@@ -18,7 +18,7 @@ here="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
 function replace_acceptable_years() {
     # this needs to replace all acceptable forms with 'YEARS'
-    sed -e 's/2017-20[12][890]/YEARS/g' -e 's/2019/YEARS/g' -e 's/2020/YEARS/g'
+    sed -e 's/2017-20[12][890]/YEARS/g' -e 's/2019/YEARS/g' -e 's/2020/YEARS/g' -e 's/2021/YEARS/g'
```
Comment (glbrntt) [inline scripts/soundness.sh:21]: I'm surprised we got this far into the year without this change!

**Review by glbrntt — APPROVED** [review]
Was `FuzzTesting/FailCases/.empty` meant to be checked in? Looks good otherwise.

**Comment by Lukasa** [comment]
Yeah, I checked `.empty` in on purpose to make the directory exist. Git doesn't create empty directories on clone.

File: FuzzTesting/Sources/FuzzHTTP2/main.swift (hunk 1 above)
Comment (Lukasa) [reply FuzzTesting/Sources/FuzzHTTP2/main.swift:20]: Done.

File: NOTICES.txt [hunk 3]
```diff
@@ -0,0 +1,17 @@
+
```
Comment (glbrntt) [inline NOTICES.txt:1]: There's already a NOTICE.txt, we should merge them

Comment (Lukasa) [reply NOTICES.txt]: Good catch!

**Pull request merged**